src/spox/_attributes.py

```diff
@@ -188,37 +202,98 @@ def maybe(
 class AttrFloat32s(_AttrIterable[float]):
     _attribute_proto_type_int = AttributeProto.FLOATS
 
-    def _to_onnx_deref(self, key: str) -> AttributeProto:
+    def _to_onnx_deref(self, key: str, only_typecheck: bool = False) -> AttributeProto:
         # ensure values are all floats
         values = [float(v) for v in self.value]
-        return make_attribute(key, values, attr_type=AttributeProto.FLOATS)
+        return (
```

Comment:
The difference in behaviour for list attributes should probably be handled in the base class, AttrIterable.

Reply:
How would you handle the custom behavior in some child classes such as:

https://github.com/Quantco/spox/blob/2c3d882156d4186d2f0d1cda99e27ea0feaf2150/src/spox/_attributes.py#L207
https://github.com/Quantco/spox/blob/2c3d882156d4186d2f0d1cda99e27ea0feaf2150/src/spox/_attributes.py#L241

We rely on exactly the same values & semantics as in the call to `make_attribute` 🤔

Reply:
Overridden class attribute/class method?

Reply:
Is https://github.com/Quantco/spox/pull/122/commits/5bad27f2733397916f7a26cefcec5042736b4ae7 what you had in mind? We could also put `_transformed` in the parent, but I was worried it might be a bit confusing at first sight.